print-bgp.c

```diff
@@ -2673,47 +2683,56 @@ bgp_open_print(netdissect_options *ndo,
         GET_BE_U_2(bgp_open_header->bgpo_holdtime));
     ND_PRINT("ID %s", GET_IPADDR_STRING(bgp_open_header->bgpo_id));
     optslen = GET_U_1(bgp_open_header->bgpo_optlen);
-    ND_PRINT("\n\t  Optional parameters, length: %u", optslen);
+    if (optslen == BGP_OPEN_NON_EXT_OPT_LEN) {
```

Comment:
rfc9072 says in 2 places that the length doesn't have to be 255 - SHOULD when constructing and MUST ignore on receipt in section 2:
```
   The Non-Extended Optional Parameters Length field (Non-Ext OP Len.)
   SHOULD be set to 255 on transmission ...
   ...; it MUST be ignored on receipt once the use of the extended
   format is determined positively by inspection of the Non-Extended
   Optional Parameters Type (Non-Ext OP Type) field.
```

And in section 3,
```
   It is not considered an error to receive an OPEN message whose
   Extended Optional Parameters Length value is less than or equal to
   255.  It is not considered a fatal error to receive an OPEN message
   whose (non-extended) Optional Parameters Length value is not 255 and
   whose first Optional Parameter type code is 255 -- in this case, the
   encoding of this specification MUST be used for decoding the message.
```

I think the fix for this is to simply eliminate this if, and just check for `opsttype == BGP_OPEN_NON_EXT_OPT_TYPE_EXTENDED_LENGTH`.

Reply:
optslen can't be ignored, because you don't know in advance how to parse the message. Encoding is determined by looking in optstype, but it should be checked only if optslen is 255, otherwise you will check one byte ahead in a wrong place.

Reply:
Can you say more about how the RFC is incorrect about how to treat the length field?  What problem do you anticipate if you ignore the Non-Extended Optional Parameters Length field, in the case where the Non-Extended Optional Parameters Type identifies an extended parameter?

Reply:
Yes, we are just about the parsing here, fixed.